collector/src/main/java/com/bitmovin/analytics/utils/HttpClient.java

```diff
@@ -19,15 +22,26 @@
     private static final String TAG = "HttpClient";
     private OkHttpClient client = null;
     private Context context;
+    private BitmovinAnalyticsConfig analyticsConfig;
 
-    public HttpClient(Context context) {
+    public HttpClient(Context context, BitmovinAnalyticsConfig analyticsConfig) {
         this.context = context;
+        this.analyticsConfig = analyticsConfig;
     }
 
 
     public void post(String url, String postBody, final Callback callback) {
         if (client == null) {
-            client = new OkHttpClient();
+            if (analyticsConfig.isResendDataOnHttpTimeout()){
+                client = new OkHttpClient.Builder()
```

Comment:
Hmm.. aren't these sane settings regardless of the retry feature? Why do we need to pass the configuration in here - probably the httpClient should be used outside of this class and be injected here?

Reply:
I removed two last setups, but this clients still is trying to resend data if I put only
`.retryOnConnectionFailure(false)` and I see duplicated samples in DB and if we put additional setting for connectTimeout then it's all working as expected.